starter-code/routes/users.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const express = require('express');
+const bcrypt = require('bcrypt');
+const bcryptSalt = 10;
+
+const User = require('../models/users');
+
+const router = express.Router();
+
+router.get('/login', function (req, res, next) {
+  if (!req.session.currentUser) {
+    res.render('auth/login');
+  } else {
+    res.redirect('/');
+  }
+});
+
+router.post('/login', (req, res, next) => {
+  if (!req.session.currentUser) {
+    const username = req.body.username;
+    const password = req.body.password;
+
+    if (!username || !password) {
+      res.redirect('login');
+      return;
+    }
+
+    User.findOne({ 'username': username })
+      .then((result) => {
+        if (result) {
+          if (bcrypt.compareSync(password, result.password)) {
+            req.session.currentUser = result;
+            res.redirect('/');
+            return;
+          }
+          res.redirect('/auth/login');
+          return;
+        }
+        res.redirect('/auth/login');
+      })
+      .catch(next);
+  }
```

Comment:
If the user is logged in and tries to post to our server the browser will be waiting for a response. If you protect the route, make sure that you cover both scenarios